### PR TITLE
Added getEntryByKeyOptional and used it. Changed result of getFieldOr…

### PR DESCRIPTION
**Pull request by oscargus**
* Added `getEntryByKeyOptional` and almost managed to completely remove the use of `getEntryByKey`.
* Changed the return type of `getFieldOrAlias` to be `Optional<String>`
* Some more conversions of `getField` to `getFieldOptional`

### CODE REVIEW

**Comment by oscargus** [comment]
Since I managed to convert all calls I renamed `getEntryByKeyOptional` to `getEntryByKey` and removed the original `getEntryByKey`.

File: src/test/java/net/sf/jabref/model/entry/BibEntryTests.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public void testDefaultConstructor() {
         // we have to use `getType("misc")` in the case of biblatex mode
         Assert.assertEquals("misc", entry.getType());
         Assert.assertNotNull(entry.getId());
-        Assert.assertNull(entry.getField("author"));
+        Assert.assertFalse(entry.getFieldOptional("author").isPresent());
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/model/entry/BibEntryTests.java]: Better `assertEquals(Optional.empty(), entry.getField(....))` in this way one gets a better failure message. Same applies to the other tests, i.e. `assertEquals(Optional.of("2003-03"), emptyEntry.getFieldOrAlias("date"))` below.

Comment (oscargus) [inline src/test/java/net/sf/jabref/model/entry/BibEntryTests.java]: Makes sense.

File: src/main/java/net/sf/jabref/gui/maintable/MainTableColumn.java [hunk 2]
```diff
@@ -112,9 +112,13 @@ public Object getColumnValue(BibEntry entry) {
             if (field.equals(BibEntry.TYPE_HEADER)) {
                 content = EntryUtil.capitalizeFirst(entry.getType());
             } else {
-                content = entry.getFieldOrAlias(field);
-                if (database.isPresent() && "Author".equalsIgnoreCase(columnName) && (content != null)) {
-                    content = database.get().resolveForStrings(content);
+                Optional<String> newContent = entry.getFieldOrAlias(field);
+                if (newContent.isPresent()) {
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/gui/maintable/MainTableColumn.java]: use ifPresent

**Comment by simonharrer** [comment]
LGTM

**Pull request merged**